### PR TITLE
#391: MsgUnjail support with unit tests

### LINKED ISSUE #391: Add support for `MsgUnjail`

### PR DESCRIPTION
**Pull request by cdc-Hitesh**
Fix #391

### CODE REVIEW

File: lib/src/core/cro.ts [hunk 1]
```diff
@@ -126,6 +127,9 @@ export const CroSDK = function (configs: InitConfigurations) {
                     CommunityPoolSpendProposalV2: communityPoolSpendProposalV2(configs),
                 },
             },
+            slashing: {
+                MsgUnjail: msgUnjail(configs),
```
Comment (calvinaco) [inline lib/src/core/cro.ts:131]: Do we need to add V2 suffix?

File: lib/src/transaction/msg/slashing/MsgUnjail.spec.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        },
+        bip44Path: {
+            coinType: 1,
+            account: 0,
+        },
+        rpcUrl: '',
+    },
+});
+
+describe('Testing MsgUnjail (Unbonding)', function () {
```
Comment (calvinaco) [inline lib/src/transaction/msg/slashing/MsgUnjail.spec.ts:32]:
```suggestion
describe('Testing MsgUnjail', function () {
```

File: lib/src/transaction/msg/slashing/MsgUnjail.spec.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    fuzzyDescribe('should throw Error when options is invalid', function (fuzzy) {
+        const anyValidOptions = {
+            validatorAddress: 'tcrocncl1j7pej8kplem4wt50p4hfvndhuw5jprxxxtenvr',
+            amount: new cro.Coin('1000', Units.BASE),
```
Comment (calvinaco) [inline lib/src/transaction/msg/slashing/MsgUnjail.spec.ts:36]: valid options should not have `amount`?

File: lib/src/transaction/msg/slashing/MsgUnjail.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+export interface MsgUnjailRaw {
+    '@type': string;
+    validator_addr: string;
+}
+
+export interface Amount {
+    denom: string;
+    amount: string;
+}
+
```
Comment (calvinaco) [inline lib/src/transaction/msg/slashing/MsgUnjail.ts:20]: This is no longer needed here.

**Review by calvinaco — CHANGES_REQUESTED** [review]
Overall LGTM, some minor issues reported.

File: lib/src/transaction/msg/slashing/MsgUnjail.ts (hunk 4 above)
Comment (cdc-Hitesh) [reply lib/src/transaction/msg/slashing/MsgUnjail.ts:20]: Oh yes ! Noted

File: lib/src/transaction/msg/slashing/MsgUnjail.spec.ts (hunk 3 above)
Comment (cdc-Hitesh) [reply lib/src/transaction/msg/slashing/MsgUnjail.spec.ts:36]: Noted.

File: lib/src/core/cro.ts (hunk 1 above)
Comment (cdc-Hitesh) [reply lib/src/core/cro.ts:131]: We wished to migrate users to V2 eventually, hence V2.

Do you suggest keeping it simple without V2? @calvinaco

Comment (calvinaco) [reply lib/src/core/cro.ts:131]: I mean this is put under v2 message, but the import path and the function name do not have the `v2` keywords.

i.e. `import { msgUnjailV2 } from '../transaction/msg/v2/slashing/v2.MsgUnjail';`

so I wonder if this is expectd?

**Review by calvinaco — COMMENTED** [review]
Replied the previous question. Also the test and lint are failing.

File: lib/src/core/cro.ts (hunk 1 above)
Comment (cdc-Hitesh) [reply lib/src/core/cro.ts:131]: Oh right got it,
I didn't use the `msgUnjailV2` as this is the first time we have supported this message type.

Comment (cdc-Hitesh) [reply lib/src/core/cro.ts]: Keeping it `msgUnjailV2` for consistency.

**Review by calvinaco — APPROVED** [review]
LGTM

**Pull request merged**